model/renderNativeWindow.h

```diff
@@ -23,13 +23,28 @@
 #include <cairoSurfaceImage.h>
 
 namespace minsky
-{
-  class RenderNativeWindow: public ecolab::CairoSurface
+{  
+  class WindowInformation;
+  class RenderNativeWindow : public ecolab::CairoSurface
   {
+  private:
+    CLASSDESC_ACCESS(RenderNativeWindow); 
+    // std::shared_ptr<WindowInformation> winInfoPtr; // TODO:: cannot get things to work with this
+    WindowInformation* winInfo;
```

Comment:
As mentioned, RenderNativeWindow should be DCAS, in particular have valid copy and assignment operators. Best to arrange all members of RenderNativeWindow be DCAS, and then take the compiler generated copy and assignment operators - this is more scalable.

I think we have two options for making the winInfo member DCAS - the shared_ptr approach I outlined, and the no-op copy operations: both have their own set of issues and implications. Maybe other methods will work, but I can't think of any.

As for the 'S' part, a simple Exclude wrapper is sufficient.